pootle/apps/pootle_data/store_data.py

```diff
@@ -6,101 +6,133 @@
+            critical_checks="critical",
+            pending_suggestions="suggestions")
+        stats = {
+            v: getattr(self.context.data, k)
+            for k, v in mapping.items()}
+        stats["is_dirty"] = False
+        stats["children"] = {}
+        stats["lastaction"] = (
+            self.context.data.last_submission
+            and self.context.data.last_submission.get_submission_info()
+            or None)
+        stats["lastupdated"] = (
+            self.context.data.last_created_unit
+            and self.context.data.last_created_unit.get_last_updated_info()
+            or None)
+        return stats
+
+    def get_checks(self, **kwargs):
+        return dict(self.context.check_data.values_list("name", "count"))
+
+
+class StoreDataUpdater(DataUpdater):
+
+    related_name = "store"
+    aggregate_fields = (
+        "words",
+        "max_unit_revision",
+        "max_unit_mtime")
+
+    @property
     def store(self):
-        return self.context
+        return self.model
+
+    @property
+    def store_data_qs(self):
+        return self.store.unit_set
 
     @property
     def units(self):
         """Non-obsolete units in this Store"""
-        return self.store.unit_set.live()
+        return self.store_data_qs.filter(state__gt=OBSOLETE)
 
-    @cached_property
-    def last_created_unit(self):
-        """Last unit created in this Store"""
-        return (
-            self.units.exclude(creation_time__isnull=True)
-                      .order_by('-creation_time', '-revision').first())
+    @property
+    def aggregate_words(self):
+        return dict(
+            total_words=models.Sum(
+                Case(
+                    When(Q(state__gt=OBSOLETE)
+                         & Q(source_wordcount__gt=0),
+                         then="source_wordcount"),
+                    default=0)),
+            translated_words=models.Sum(
+                Case(
+                    When(Q(state=TRANSLATED)
+                         & Q(source_wordcount__gt=0),
+                         then="source_wordcount"),
+                    default=0)),
+            fuzzy_words=models.Sum(
+                Case(
+                    When(Q(state=FUZZY)
+                         & Q(source_wordcount__gt=0),
+                         then="source_wordcount"),
+                    default=0)))
 
-    @cached_property
-    def last_updated_unit(self):
-        """Last unit updated in this Store"""
-        return self.units.order_by('-revision', '-mtime').first()
+    @property
+    def aggregate_max_unit_revision(self):
+        return dict(max_unit_revision=Max("revision"))
 
-    @cached_property
-    def last_submission(self):
-        """Last non-UNIT_CREATE submission for this store"""
+    @property
+    def aggregate_max_unit_mtime(self):
+        return dict(max_unit_mtime=Max("mtime"))
+
+    def get_last_created_unit(self, **kwargs):
+        order_by = ("-creation_time", "-revision", "-id")
```

Comment:
I hope `revision` and `id` don't affect `(store, creation_time)` index_together.